components/camel-rest-openapi/src/main/java/org/apache/camel/component/rest/openapi/DefaultRestOpenapiProcessorStrategy.java

```diff
@@ -145,6 +149,21 @@ public void validateOpenApi(OpenAPI openAPI, PlatformHttpConsumerAware platformH
         }
     }
 
+    /**
+     * If the operation has no operationId specified, generate one based on the path and the operation method.
+     *
+     * @param  path       The path for this operation, such as /users.
+     * @param  httpMethod The operation to perform
+     * @return            A generated operation id based on the path and the operation. Slashes and braces in the path
+     *                    are replaced with placeholder characters.
+     */
+    private String generateOperationId(String path, HttpMethod httpMethod) {
+        final String sanitizedPath = path.replace('/', '.').replaceAll("[{}]", "_");
+        final String opId = GEN_OPID + httpMethod.name() + sanitizedPath;
+        LOG.warn("Generated operationId {} for path {} and method {}", opId, path, httpMethod.name());
```

Comment:
WARN is maybe too high level - the openapi spec seem to allow no operation id, and as such we should just accept this - so a DEBUG logging is maybe better